source/grammar/openpulseParser.g4

```diff
@@ -0,0 +1,70 @@
+/***** ANTLRv4  grammar for OpenPulse. *****/
+// This grammar is OpenPulse extension for QOpenQASM3.0
+// We introduce several new types for OpenPulse.
+// We override several nodes in qasm3 to consume the new types.
+// The OpenPulse grammar can be used inside cal or defcal blocks.
+
+parser grammar openpulseParser;
+import qasm3Parser;
+
+options {
+    tokenVocab = openpulseLexer;
+}
+
+/**** OpenQASM3.0 overrides ****/
+defcalStatement: DEFCAL Identifier (LPAREN argumentDefinitionList? RPAREN)? hardwareQubitList returnSignature? LBRACE statement* RBRACE;
```

Comment:
Per my interpretation of the OQ 3 spec, the `defcal` statement should be valid OQ 3, only the _block_ should be in the calibration grammar, so we shouldn't need to do this bit, I think.  `defcal` blocks presumably shouldn't be able to take or return `waveform` (because it's not an OQ 3 type), and the qubit lists parts should be handled by the OQ 3 parser.

There's definitely bugs in the reference ANTLR one surrounding this, but I also think there's a little bit of mixing in the spec between where exactly calibration-grammar-specific constructs are allowed.  I opened #367 about that.

Reply:
So do you envisioned that we construct OpenPulse parser/ast, for the body of `cal` and `defcal`, completely independent of OpenQASM?

Reply:
For tracking of discussion for my own review see - https://github.com/openqasm/openqasm/pull/352#issuecomment-1179209502